src/connectors/impls/gbq/writer/sink.rs

```diff
@@ -12,7 +12,11 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use crate::connectors::google::AuthInterceptor;
+#[cfg(not(test))]
```

Comment:
I think we can avoid all the cfg test in this file if we generalize `GbqSink` and `AuthInterceptor ` over `TokenProvider` and implement a default method, then during normal operations we can use `GbqSink<GouthTokenProvider>` and during tests `GbqSink<TestTokenProvider>`

Reply:
I did that without generalizing, just by using a type alias, I think that's a simpler solution, let me know what you think :)

Reply:
Not really to be honest, there are still 10 compile-time conditionals in the code which makes the whole thing quite hard to read.

Reply:
Hmm... I only see a few right now, almost all of them making the TestTokenProvider only compile in tests 🤔. Can you point at the specific problematic lines? Maybe I'm missing something

Reply:
Sure,

25, 27, 31, 36, 49, 66 in google.rs

all of those can be avoided, the Test porovider could move into `mod test` entirely.

I'm not a fan of the type alias either, as compile time switches are, when avoidable, always an addition in complexity that makes code harder to reason about, read, and maintain. We are not writing C where we don't have any other options.

Reply:
donezo, unless I missed anything, this should be done now :)